File: advent-of-code-2024/cpp/day1.cpp
#include<bits/stdc++.h>
using namespace std;
const string sample = "..\\samples\\day1.txt";

vector<string> splitString(string &s, char d){
    vector<string> tokens;
    stringstream ss(s);
    string token;
    while(getline(ss, token, d)){
        tokens.push_back(token);
    }
    return tokens;
}

// Function that appends a string to another using a reference
void appendString(std::string& base, const std::string toAppend) {
    base += toAppend;
}


int read(string &value, string path){
    ifstream file(sample);  // Open the file
    if (!file.is_open()) {
        cerr << "Unable to open file.\n";
        return 1;  // Return an error code
    }

    string line;
    
    while (getline(file, line)) {  // Read line by line
        // cout << line << endl;
        line +="\n";
        appendString(value, line);

    }

    file.close();  // Close the file
    return 0;
}


int main() {
    string sample1 = "";
    read(sample1, sample);
    vector<string> lines = splitString(sample1, '\n');
    
   vector<int> left_list;
   vector<int> right_list;
    for(auto it = lines.begin(); it != lines.end(); ++it){
        // cout <<*it << endl;
        auto something = splitString(*it, ' ');
        for(auto it2 = something.begin(); it2 != something.end(); ++it2){
            cout << "abcd "<<*it2 << endl;
        }

        // left_list.push_back(stoi(something.at(0)));
        // right_list.push_back(stoi(something.at(1)));
    }

   

   
    return 0;
}

File: advent-of-code-2024/python/day1.py
day1 = ""
with open("../inputs/day1.txt", 'r') as o:
    day1 = o.read()

sample = """3   4
4   3
2   5
1   3
3   9
3   3"""
s1 = day1.split("\n")
a = []
b = []
for s in s1:
    n = s.split("   ")
    if len(n) != 0:
        a.append(n[0])
        b.append(n[1])

a.sort()
b.sort()
n = 0
for i in range(0, len(a)):
    n += abs(int(a[i]) - int(b[i]))

print("Part One Answer is: ", n)

c = {}


def get_frequency(x: int) -> int:
    m = 0
    for i in b:
        if x == int(i):
            m = m+1
    return m
# for i in b:
#     if c.get(i) is None:
#         c[i] = 0
#     else:
#         c[i] = c[i] + 1
m = 0

for i in a:
    # if c.get(i) is None:
    #     continue
    m = m + int(i)*get_frequency(int(i))
print(m)
